From 6a82ba35266ec0530c34901542a61ae02b640213 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Sun, 6 Dec 2015 16:31:07 -0800
Subject: [PATCH 5/6] ODROID-C: Do not return success from probe if DAC is
 disabled

Returning success early like this will leave a device in the system
which doesn't have any drvdata. Routines in soc-audio will expect
the drvdata to be initialized, and WILL dereference it without checking.
---
 sound/soc/aml/m8/odroid_dac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/aml/m8/odroid_dac.c b/sound/soc/aml/m8/odroid_dac.c
index dfcefc5..57b1137 100644
--- a/sound/soc/aml/m8/odroid_dac.c
+++ b/sound/soc/aml/m8/odroid_dac.c
@@ -266,7 +266,7 @@ static int odroid_audio_probe(struct platform_device *pdev)
     int ret = 0;
 
     if(!enable_audiodac)
-        return 0;
+        return -ENODEV;
 
 #ifdef CONFIG_USE_OF
     p_odroid_audio = devm_kzalloc(&pdev->dev,
-- 
2.1.0

